sycl/test/virtual-functions/diagnostics-negative.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: %clangxx -fsycl -fsycl-device-only -c -Xclang -verify %s -o %t.ignored
+//
+// This test is intended to check that correct diagnostics are emitted if
+// kernel attempts to perform virtual calls, but wasn't submitted with the right
+// properties.
+
+#include <sycl/sycl.hpp>
+
+namespace oneapi = sycl::ext::oneapi::experimental;
+
+class Base {
+public:
+  SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(oneapi::indirectly_callable<>)
+  virtual void foo() const {}
+
+  virtual void bar() const {}
+};
+
+class Derived : public Base {
+public:
+  SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(oneapi::indirectly_callable<>)
+  void foo() const override {}
+
+  void bar() const override {}
+};
+
+void foo(Base *Ptr) { // expected-note {{performed by function 'foo(Base*)'}}
+  Ptr->foo();
+}
+
+void bar(Base *Ptr) { // expected-note {{performed by function 'bar(Base*)'}}
+  Ptr->bar();
+}
+
+int main() {
+  sycl::queue q;
+
+  Base *Ptr = sycl::malloc_device<Derived>(1, q);
```

Comment:
I know the test is compile-only, but should we free this ptr for correctness?

Reply:
Same for `diagnostics-positive.cpp`.

Reply:
Added `free` in e8be5de14e5e15f87ffc24524e4fae29c2e9f391